ocaml/idl/datamodel_cluster_host.ml

```diff
@@ -111,9 +111,21 @@ let t =
            ~default_value:(Some (VRef null_ref)) "Reference to the PIF object"
        ; field ~qualifier:StaticRO ~lifecycle ~ty:Bool "joined"
            ~default_value:(Some (VBool true))
-           "Whether the cluster host has joined the cluster"
-         (* TODO: add `live` member to represent whether corosync believes that this
-                  cluster host actually is enabled *)
+           {|Whether the cluster host has joined the cluster. Contrary to \
+            `enabled`, a host that is not joined is not considered a member of \
```

Comment:
Could the coordinator be not joined the cluster? This is a configuration, rather than a state?

Reply:
I think it is possible for the pool coordinator to not be in the cluster. In that case these fields will reflect the view from the pool coordinator, i.e. whehther the pool coordinator thinks another cluster host is alive, even if the coor is not in the majority quorum.

Of course, this information is probably inaccurate, but there isn't much we can do if the coordinator is not in the cluster. I guess we could disable the watcher when this is the case?

Reply:
If the pool coordinator is not in the cluster, would it be able to get the update from corosync?

Reply:
I think so. Even if the coordinator is not in the majority quorum, it should still be able to contact corosync, but get back info on what it thinks the quorum state is.

Reply:
Don't know if it is actually a good idea to do that, we can either say that these fields always reflect the view of the coordinator (which implies that when the coordinator is not in the majority quorum, its view will be inaccurate), or we can disable this update when the coordinator is not in the majority quorum.

Reply:
Re the config vs state: I think we are using the state/config a bit interchangbly. I guess you probably mean that a config is a corosync config file, in which case I think we do change the config file each time when we update the joined field.

Reply:
The `config` means a host is required (by users) to join into a cluster. While the `state` means it is required but is failing due to e.g. networking issue.
(Sorry I'm not familiar with all the concepts in cluster)
If the coordinator is not required to be in a cluster, I don't understand why its corosync can get the information from the cluster.